src/operator/nn/dropout-inl.h

```diff
@@ -228,11 +301,27 @@ class DropoutOp {
         if (!MKLForward(s, pgen, this->pkeep_, in_data, out_data)) {
           const TBlob &mask = out_data[dropout::kMask];
           CHECK(req[dropout::kOut] != kAddTo);
-          LaunchRNG<DropoutKernel, xpu>(s, pgen, out.Size(),
-                                        out.dptr<DType>(),
-                                        mask.dptr<DType>(),
-                                        in_data[dropout::kData].dptr<DType>(),
-                                        this->pkeep_);
+          // initialize the mask
+          LaunchRNG<BernoulliKernel, xpu>(s, pgen, out.Size(),
+                                          mask.dptr<DType>(),
+                                          this->pkeep_);
+          if (req[0] != kNullOp) {
+            // broardcast mul
+            TShape new_lshape, new_rshape, new_oshape;
+            int ndim = BinaryBroadcastShapeCompact(in_data[dropout::kData].shape_,
+                                                   mask.shape_, out.shape_,
+                                                   &new_lshape, &new_rshape, &new_oshape);
+            BROADCAST_NDIM_SWITCH(ndim, NDim, {
+              mshadow::Shape<NDim> oshape = new_oshape.get<NDim>();
+              mshadow::Shape<NDim> lstride = mxnet_op::calc_stride(new_lshape.get<NDim>());
+              mshadow::Shape<NDim> rstride = mxnet_op::calc_stride(new_rshape.get<NDim>());
+              mxnet_op::Kernel<mxnet_op::binary_broadcast_kernel<NDim, DType,
+                               mshadow_op::mul>, xpu>::
+              template LaunchEx(s, new_oshape.Size(), req[0], lstride, rstride, oshape,
+              in_data[dropout::kData].dptr<DType>(),
+              mask.dptr<DType>(), out.dptr<DType>());
+            });
+          }
         }
```

Comment:
if MKL enables, the broadcast will not happen?

Reply:
I haven't updated the MKL code for variational dropout (enabling axes). I need help with MKL

Reply:
for non default axes, you can fall back to this op instead of the MKL op

Reply:
Thx @cjolivier01 . I added the condition check here https://github.com/apache/incubator-mxnet/pull/9931/files#diff-4aea2cc24c0bb4e8e48face9faf4aa26R249